docs/proposals/proposal-audit-log.md

```diff
@@ -0,0 +1,256 @@
+# Audit log backend part
+
+### Purpose of audit log
+
+The audit log is a feature that allows users to track changes made to their organization, projects,
+schemas, and other resources.
+
+
+## Implementation
+
+Proposed implementation is to use a single Clickhouse table to store all events. The table will have
+the following schema:
+
+
+```sql
+CREATE TABLE audit_log (
+  event_time DateTime DEFAULT now(),
+  user_id UUID,
+  user_email STRING,
+  organization_id UUID,
+  project_id UUID,
+  project_name STRING,
+  target_id UUID,
+  target_name STRING,
+  schema_version_id UUID,
+  event_action STRING NOT NULL,
+  event_details JSON,
+  INDEX idx_user_id user_id TYPE set(0) GRANULARITY 64,
+  INDEX idx_user_email user_email TYPE set(0) GRANULARITY 64,
+  INDEX idx_organization_id organization_id TYPE set(0) GRANULARITY 64,
+  INDEX idx_project_id project_id TYPE set(0) GRANULARITY 64,
+  INDEX idx_target_id target_id TYPE set(0) GRANULARITY 64,
+) ENGINE = MergeTree ()
+ORDER BY event_time
+TTL timestamp + INTERVAL 3 MONTH;
+```
+Data in clickhouse would be append only. We would never manually delete them, we would rely on TTL removing old rows after X months
+Of course we could make the interval configurable if necessary.
+
+Audit log is simple class with one method that inserts a single row into the clickhouse table:
+
+```ts
+const { createClient } = require('@clickhouse/client');
+const clickhouse = createClient()
+
+type AuditLogEvent = {
```

Comment:
We need to be support all events in a backwards compatible manner, so I guess we will need something more strict than `string` for an event type and for the details, something to avoid "reading 'foo' of undefined" errors.

Reply:
changed `eventAction` to be AuditLogEventAction.
That can be a generated enum from the GQL schema.